### PR TITLE
Refactor DCC decoder

### PR DESCRIPTION
**Pull request by balazsracz**
Splits the Tiva DCC Decoder into a shared driver and a hardware-specific module.
This is in preparation to port the DCC decoder driver to other MCUs.

### CODE REVIEW

File: src/freertos_drivers/common/DccDecoder.hxx [hunk 1]
```diff
@@ -24,34 +24,30 @@
...
+
+  This driver exports a filesystem device node, which will be readable for
+  decoded DCC packets (read into struct DCCPacket). All writes fail.
+
+  The additional feature supported by this device driver is that it is able to
+  tell a RailcomDriver when the railcom cutout is supposed to start, when we're
+  in the middle and when it is over. This is necessary for the correct
+  functionality of the railcom driver.
 
+  @todo describe usage.
```
Comment (bakerstu) [inline src/freertos_drivers/common/DccDecoder.hxx:50]: Should this still be a todo item? It looks like you are describing usage.

File: src/freertos_drivers/common/DccDecoder.hxx [hunk 2]
```diff
@@ -200,67 +196,27 @@ private:
         MAP_TimerEnable(HW::TIMER_BASE, HW::SAMPLE_TIMER);
         }*/
 
+    /// Module::set_cap_timer_delay_usec(int usec)
```
Comment (bakerstu) [inline src/freertos_drivers/common/DccDecoder.hxx:199]: Please verify that this syntax gets picked up by Doxygen to associate the comment with DccDecoder::Module::set_cap_timer_delay_usec().

File: src/freertos_drivers/common/DccDecoder.hxx (hunk 1 above)
Comment (balazsracz) [reply src/freertos_drivers/common/DccDecoder.hxx:50]: Actually all this description was outdated stuff that should have been on the Tiva side. Wrote new description.

File: src/freertos_drivers/common/DccDecoder.hxx (hunk 2 above)
Comment (balazsracz) [reply src/freertos_drivers/common/DccDecoder.hxx:199]: No it isn't picked up. I removed these. These docu comments are in the TivaDccDecoder, and I added a link to point there.

**Pull request merged**